CanvasPlusPlayground/Common/Network/API Requests/GetPagesRequest.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  GetPagesRequest.swift
+//  CanvasPlusPlayground
+//
+//  Created by Max Ko on 2/21/25.
+//
+
+import Foundation
+
+struct GetPagesRequest: CacheableArrayAPIRequest {
+    typealias Subject = PageAPI
+
+    let courseId: String
+
+    var path: String { "courses/\(courseId)/pages" }
+
+    var queryParameters: [QueryParameter] {
+        [
+            ("sort", sort),
+            ("order", order),
+            ("search_term", searchTerm),
+            ("published", published),
+            ("per_page", perPage)
+        ]
+        + include.map { ("include[]", $0.rawValue) }
+    }
+
+    // MARK: Query Params
+    let sort: String?   // Allowed values: title, created_at, updated_at
+    let order: String?  // Allowed values: asc, desc
+    let searchTerm: String?
+    let published: Bool?
+    let include: [Include]
+    let perPage: Int
+
+    init(
+        courseId: String,
+        sort: String? = nil,
+        order: String? = nil,
+        searchTerm: String? = nil,
+        published: Bool? = nil,
+        include: [Include] = [],
+        perPage: Int = 50
+    ) {
+        self.courseId = courseId
+        self.sort = sort
+        self.order = order
+        self.searchTerm = searchTerm
+        self.published = published
+        self.include = include
+        self.perPage = perPage
+    }
+
+    var requestId: String? { courseId }
+    var requestIdKey: ParentKeyPath<Page, String?> { .createWritable(\.courseID) }
```

Comment:
If `courseID` is something given by the API, and not a custom property that you add as a part of `Page`, this just needs to be `createReadable`
```suggestion
    var requestIdKey: ParentKeyPath<Page, String?> { .createReadable(\.courseID) }
```

Reply:
its not part of the API, it's something I have to pass to a `Page` from Courses

Reply:
Ah ok, then you can ignore this comment